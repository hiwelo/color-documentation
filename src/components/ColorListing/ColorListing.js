import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import defaultOptions from '../../config/defaultOptions';
import ColorPalette from '../ColorPalette';

const componentClass = `${defaultOptions.styles.classPrefix}listing`;

const StyledListingContainer = styled.section`
  font-family: ${props => props.styles.typography.stack};
`;

const StyledMainHeading = styled.h1``;

const ColorListing = props => {
  const palettes = Object.keys(props.colors);

  return (
    <StyledListingContainer className={`${componentClass}`} styles={props.options.styles}>
      {React.isValidElement(props.options.content.mainHeading.element) ? () => {
        const Option = React.cloneElement(props.options.content.mainHeading.element, {
          children: props.options.content.mainHeading.content,
        });

        return Option;
      } : (
        <StyledMainHeading className={`${componentClass}__heading`} as={props.options.content.mainHeading.markup}>
          {props.options.content.mainHeading.content}
        </StyledMainHeading>
      )}
      {palettes.map((palette, index) => (
        <ColorPalette palette={props.colors[palette]} name={palette} options={props.options} key={index} />
      ))}
    </StyledListingContainer>
  );
};

ColorListing.propTypes = {
  /**
   * Specify a listing of all colours and palettes
   */
  colors: PropTypes.object,

  /**
   * Specify this component options
   */
  options: PropTypes.object,
};

ColorListing.defaultProps = {
  colors: {},
  options: defaultOptions,
}

export default ColorListing;
