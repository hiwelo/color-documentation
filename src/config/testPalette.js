const testPalette = {
  generic: {
    black: "#000",
    white: "#fff"
  }
};

export default testPalette;
