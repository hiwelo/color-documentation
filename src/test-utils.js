import { render } from '@testing-library/react';

/**
 * This file is compiling all useful methods
 * and consts used by unit tests.
 */

const queryByTestId = (testId) => {
  return document.querySelector(`[data-testid="${testId}"]`);
};

const queryParentByTestId = (testId) => {
  return queryByTestId(testId).parentElement;
};

const getTag = (element) => {
  return element.tagName.toLowerCase();
}

const doSnapshotTest = (DOMNode) => {
  const { container } = render(DOMNode);

  expect(container.firstChild).toMatchSnapshot();
};

export { queryByTestId, queryParentByTestId, getTag, doSnapshotTest };
