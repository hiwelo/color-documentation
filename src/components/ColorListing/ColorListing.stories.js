import React from 'react';
import { storiesOf } from '@storybook/react';
import ColorListing from './ColorListing';

const colors = {
  generic: {
    black: '#000',
    white: '#fff',
  },
};

storiesOf('ColorListing', module)
  .add('overview', () => (
    <ColorListing colors={colors} />
  ))
;