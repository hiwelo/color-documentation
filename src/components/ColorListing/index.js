import ColorListing from './ColorListing';

export default ColorListing;
