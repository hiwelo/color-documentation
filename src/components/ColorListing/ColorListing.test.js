import React from "react";
import { render } from "@testing-library/react";
import ColorListing from "./ColorListing";
import { doSnapshotTest } from "../../test-utils.js";
import testPalette from "../../config/testPalette";

describe("ColorListing", () => {
  it("should render without crashing", () => {
    render(<ColorListing colors={testPalette} />);
  });

  it("should match snapshot", () => {
    doSnapshotTest(<ColorListing colors={testPalette} />);
  });
});
