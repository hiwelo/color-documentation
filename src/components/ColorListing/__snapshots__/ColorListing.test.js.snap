// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorListing should match snapshot 1`] = `
<section
  class="sc-htpNat KtFsv"
>
  <h1
    class="sc-bxivhb bUZGRr"
  >
    Colour documentation
  </h1>
  <article
    class="sc-bdVaJa bDWFJH"
  >
    <h2
      class="sc-bwzfXH uTxCW"
    >
       generic palette
    </h2>
  </article>
</section>
`;
